### PR TITLE
feat: deploy implementation contracts for v1_2_1 - PE-50

### PR DESCRIPTION
**Pull request by koderholic**
## Description

Changes included:
- updated deploy scripts to for v1_2_1 implementation deployment, and proxy upgrade.

## Notice

- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/liquid-collective/liquid-collective-protocol/pulls) for the same update/change?
- [x] Have you assigned this PR to yourself?
- [x] Have you added at least 1 reviewer?
- [ ] Have you updated the official documentation?
- [x] Have you added sufficient documentation in your code?
- [ ] Have you added relevant tests to the official test suite?

## Pull Request Type

- [ ] 💫 New Feature (Breaking Change)
- [x] 💫 New Feature (Non-breaking Change)
- [ ] 🛠️ Bug fix (Non-breaking Change: Fixes an issue)
- [ ] 🕹️ Chore (Non-breaking Change: Doc updates, pkg upgrades, typos, etc..)

## Testing

- [x] Have you tested this code with the official test suite?
- [x] Have you tested this code manually?

### CODE REVIEW

**Review by jdiuwe — DISMISSED** [review]

**Review by iamsahu — CHANGES_REQUESTED** [review]
The scripts look good. Just a small change to make things easier in future upgrades.

File: deploy/devHolesky/08_deployment_v1_2_1.ts [hunk 1]
```diff
@@ -6,56 +6,56 @@ import { verify } from "../../scripts/helpers";
 const func: DeployFunction = async function ({ deployments, getNamedAccounts }: HardhatRuntimeEnvironment) {
   const { deployer } = await getNamedAccounts();
 
-  const allowlistDeployment = await deployments.deploy("AllowlistV1_Implementation_1_2_0", {
+  const allowlistDeployment = await deployments.deploy("AllowlistV1_Implementation_1_2_1", {
```
[the remark this answers is not in the document]
Comment (koderholic) [reply deploy/devHolesky/08_deployment_v1_2_1.ts:9]: Ok, so we use the same script for future deployments, just change of version. Ok, good.

**Review by iamsahu — APPROVED** [review]
LGTM!

**Review by juliaaschmidt — APPROVED** [review]
LGTM

**Pull request merged**